04_numeronym/madhavansu/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+func numeronyms(vals ...string) []string {
+	s := make([]string, len(vals))
+	for k, val := range vals {
+		val := []rune(strings.Trim(val, "\n\t\\ "))
+		length := len(val)
+		if length > 3 {
+			s[k] = fmt.Sprint(string(val[0]), length-2, string(val[length-1]))
```

Comment:
This works fine. An interesting alternative I thought of
`s[k] = fmt.Sprintf("%c%d%c", val[0], length-1, val[length-1])`
Advantage: removes casts
Disadvantage: runtime cost of string formatting

Reply:
Updated.